axon-server/administration/clustering.md

```diff
@@ -1,55 +1,121 @@
+
+To use the cluster template feature, all you need to do is define a valid cluster template yml file. If this file is present on a fresh Axon Server startup, it will automatically be picked up and the cluster will be configured accordingly.
+
+{% hint style="info" %}
+Each cluster node needs to have the cluster template yml file copy. Each node will read this file, find its own configuration and configure itself.
+{% endhint %}
+
+Default path from which Axon Server reads configuration is `./cluster-template.yml`
+
+You can override this path anytime by setting Axon Server property: `axoniq.axonserver.clustertemplate.path:/mypath/cluster-template.yml`
+
+#### Configuration
+
+Below you can find an example of a basic cluster setup: the \_admin and default contexts are in separate replication nodes, replicated across all nodes that are marked as primary.
+
+```yaml
+axoniq:
+  axonserver:
+    cluster-template:
+      first: axonserver-enterprise-1
+      replicationGroups:
+      - name: _admin
+        roles:
+        - node: axonserver-enterprise-1
+          role: PRIMARY
+        - node: axonserver-enterprise-2
+          role: PRIMARY
+        - node: axonserver-enterprise-3
+          role: PRIMARY
+        contexts:
+        - name: _admin
+      - name: default
+        roles:
+        - node: axonserver-enterprise-2
+          role: PRIMARY
+        - node: axonserver-enterprise-3
+          role: PRIMARY
+        - node: axonserver-enterprise-1
+          role: PRIMARY
+        contexts:
+        - name: default
+      applications: []
+      users: []
+```
+
+![Cluster overview after default configuration is applied](https://github.com/domaincomponents/reference-guide-feature-dev/tree/615f89fe44783bf80563113cb22ef7d039b8f38b/.gitbook/assets/cluster-template-default-configuration.png)
+
+_Cluster overview after default configuration is applied_
+
+#### Export/Generator
+
+In order to avoid mistakes while writing a cluster configuration file, we have implemented an export button that will generate a cluster template file based on current setup.
+
+![Cluster Template export button location](https://github.com/domaincomponents/reference-guide-feature-dev/tree/615f89fe44783bf80563113cb22ef7d039b8f38b/.gitbook/assets/cluster-template-export-button.png)
+
+_Location of export button at Settings page_
+
+**Recommended mechanism - Creating an advanced cluster setup**
+
+* Start a fresh Axon Server setup \(use basic cluster template setup mentioned above\).
+* Configure a cluster via the UI, by creating users, applications, replication groups and contexts.
+* Use the export button located at "Settings -&gt; Configuration" panel  to download the current cluster configuration.
+* Replace the basic cluster template with the newly exported cluster template configuration.
+
+{% hint style="info" %}
+Use export button from main - leader node. Leader node always contains complete cluster configuration, while it might happen that followers don't.
```

Comment:
export from any admin node to ensure that the configuration file contains all the relevant information.